src/theory/quantifiers/ematching/candidate_generator.h

```diff
@@ -208,6 +212,26 @@ class CandidateGeneratorConsExpand : public CandidateGeneratorQE
   bool isLegalOpCandidate(Node n) override;
 };
 
+/**
+ * Candidate generator for selector applications, which considers both
+ * internal terms corresponding to correctly and incorrectly applied selectors.
+ */
+class CandidateGeneratorSelector : public CandidateGeneratorQE
+{
+ public:
+  CandidateGeneratorSelector(QuantifiersEngine* qe, Node mpat);
+  /** reset */
+  void reset(Node eqc) override;
+  /** get next candidate */
+  Node getNextCandidate() override;
+
+ protected:
+  /** the selector operator */
+  Node d_selOp;
+  /** the UF operator */
+  Node d_ufOp;
```

Comment:
Seems a bit wasteful to keep these duplicates as nodes. Do you think it's worth it to use TNodes for this?

Reply:
This class should only be constructed a handful of times per benchmark.  It's not important to do TNode here.